src/controllers/TalksController.php

```diff
@@ -548,7 +548,7 @@ public function setSpeakerForTalk(Request $request, PDO $db)
 
         if ($claim === false) {
             throw new Exception("No speaker matching that name found", 422);
-        } elseif ($claim['speaker_id'] != null) {
+        } elseif ($claim['speaker_id'] != null && $claim['speaker_id'] != 0) {
```

Comment:
It should, but for some reason (that I've not been able to figure out) it's not.... with it was just != null, rows with a speaker_id of 0 were throwing the 'already claimed' exception, and putting in the != 0 worked...

I'm at a loss to explain why, but that's what it's doing!